From e4c463cfd433271cd90e95337a6c01d166052271 Mon Sep 17 00:00:00 2001
From: Nikolay Orlyuk <virkony@gmail.com>
Date: Sat, 7 Jul 2012 20:53:25 +0300
Subject: [PATCH 1/2] Allow /usr/lib64 override (bug259)

Many linux distributives uses /usr/lib64 for amd64 libraries and
/usr/lib32 for x86. To support those this change allows to override
/usr/lib with standart libdir var in autoconf.

See http://bugs.mercury.csse.unimelb.edu.au/view.php?id=259
---
 scripts/Mmake.vars.in | 28 ++++++++++++++++------------
 1 file changed, 16 insertions(+), 12 deletions(-)

diff --git a/scripts/Mmake.vars.in b/scripts/Mmake.vars.in
index c21b166..2c0ac46 100644
--- a/scripts/Mmake.vars.in
+++ b/scripts/Mmake.vars.in
@@ -682,26 +682,30 @@ else
     DESTDIR_AND_SLASH=$(DESTDIR)/
 endif
 
-INSTALL_PREFIX		= $(DESTDIR_AND_SLASH)@prefix@
+prefix = @prefix@
+exec_prefix = @exec_prefix@
+libdir = @libdir@
+
+INSTALL_PREFIX		= $(DESTDIR_AND_SLASH)$(prefix)
 INSTALL_BINDIR		= $(INSTALL_PREFIX)/bin
-INSTALL_LIBDIR		= $(INSTALL_PREFIX)/lib/mercury
+INSTALL_LIBDIR		= $(DESTDIR_AND_SLASH)$(libdir)/mercury
 INSTALL_INFO_DIR	= $(INSTALL_PREFIX)/info
-INSTALL_DVI_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_TEXT_DIR	= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PS_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PDF_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
+INSTALL_DVI_DIR		= $(INSTALL_LIBDIR)/doc
+INSTALL_TEXT_DIR	= $(INSTALL_LIBDIR)/doc
+INSTALL_PS_DIR		= $(INSTALL_LIBDIR)/doc
+INSTALL_PDF_DIR		= $(INSTALL_LIBDIR)/doc
 INSTALL_MAN_DIR		= $(INSTALL_PREFIX)/man
-INSTALL_HTML_DIR	= $(INSTALL_PREFIX)/lib/mercury/html
-INSTALL_MDB_DOC_DIR	= $(INSTALL_PREFIX)/lib/mercury/mdb
-INSTALL_ELISP_DIR	= $(INSTALL_PREFIX)/lib/mercury/elisp
+INSTALL_HTML_DIR	= $(INSTALL_LIBDIR)/html
+INSTALL_MDB_DOC_DIR	= $(INSTALL_LIBDIR)/mdb
+INSTALL_ELISP_DIR	= $(INSTALL_LIBDIR)/elisp
 INSTALL_CGI_DIR		= $(DESTDIR_AND_SLASH)@CGIDIR@
 
-FINAL_INSTALL_PREFIX	= @prefix@
+FINAL_INSTALL_PREFIX	= $(prefix)
 FINAL_INSTALL_BINDIR 	= $(FINAL_INSTALL_PREFIX)/bin
-FINAL_INSTALL_LIBDIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury
+FINAL_INSTALL_LIBDIR	= $(libdir)/mercury
 FINAL_INSTALL_INFO_DIR	= $(FINAL_INSTALL_PREFIX)/info
 FINAL_INSTALL_MAN_DIR	= $(FINAL_INSTALL_PREFIX)/man
-FINAL_INSTALL_ELISP_DIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury/elisp
+FINAL_INSTALL_ELISP_DIR	= $(FINAL_INSTALL_LIBDIR)/elisp
 FINAL_INSTALL_CGI_DIR	= @CGIDIR@
 
 # You should not need to override anything below here
-- 
2.1.2


From 8742878924af95d7918433d92ce6a45c25729d47 Mon Sep 17 00:00:00 2001
From: Nikolay Orlyuk <virkony@gmail.com>
Date: Sat, 7 Jul 2012 23:36:07 +0300
Subject: [PATCH 2/2] Add man, info and lisp dirs configuring (bug259)

See http://bugs.mercury.csse.unimelb.edu.au/view.php?id=259
---
 configure.ac          |  3 +++
 scripts/Mmake.vars.in | 19 +++++++++++++------
 2 files changed, 16 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0117d1b..158ca46 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1182,6 +1182,9 @@ AC_SUBST(LINK_OPT_SEP)
 AC_SUBST(FIX_PATH_FOR_CC)
 AC_SUBST(CYGPATH)
 
+lispdir='${datarootdir}/emacs/site-lisp'
+AC_SUBST(lispdir)
+
 #-----------------------------------------------------------------------------#
 # Check for `-lm': some systems, e.g. MacOS X (Darwin), don't have it.
 # The result of this check may be overridden below.
diff --git a/scripts/Mmake.vars.in b/scripts/Mmake.vars.in
index 2c0ac46..dc11640 100644
--- a/scripts/Mmake.vars.in
+++ b/scripts/Mmake.vars.in
@@ -682,30 +682,37 @@ else
     DESTDIR_AND_SLASH=$(DESTDIR)/
 endif
 
+# autoconf provided vars
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 libdir = @libdir@
+datarootdir = @datarootdir@
+mandir = @mandir@
+infodir = @infodir@
+lispdir = @lispdir@
+
+# actual vars used for installation
 
 INSTALL_PREFIX		= $(DESTDIR_AND_SLASH)$(prefix)
 INSTALL_BINDIR		= $(INSTALL_PREFIX)/bin
 INSTALL_LIBDIR		= $(DESTDIR_AND_SLASH)$(libdir)/mercury
-INSTALL_INFO_DIR	= $(INSTALL_PREFIX)/info
+INSTALL_INFO_DIR	= $(infodir)
 INSTALL_DVI_DIR		= $(INSTALL_LIBDIR)/doc
 INSTALL_TEXT_DIR	= $(INSTALL_LIBDIR)/doc
 INSTALL_PS_DIR		= $(INSTALL_LIBDIR)/doc
 INSTALL_PDF_DIR		= $(INSTALL_LIBDIR)/doc
-INSTALL_MAN_DIR		= $(INSTALL_PREFIX)/man
+INSTALL_MAN_DIR		= $(DESTDIR_AND_SLASH)$(mandir)
 INSTALL_HTML_DIR	= $(INSTALL_LIBDIR)/html
 INSTALL_MDB_DOC_DIR	= $(INSTALL_LIBDIR)/mdb
-INSTALL_ELISP_DIR	= $(INSTALL_LIBDIR)/elisp
+INSTALL_ELISP_DIR	= $(DESTDIR_AND_SLASH)$(lispdir)
 INSTALL_CGI_DIR		= $(DESTDIR_AND_SLASH)@CGIDIR@
 
 FINAL_INSTALL_PREFIX	= $(prefix)
 FINAL_INSTALL_BINDIR 	= $(FINAL_INSTALL_PREFIX)/bin
 FINAL_INSTALL_LIBDIR	= $(libdir)/mercury
-FINAL_INSTALL_INFO_DIR	= $(FINAL_INSTALL_PREFIX)/info
-FINAL_INSTALL_MAN_DIR	= $(FINAL_INSTALL_PREFIX)/man
-FINAL_INSTALL_ELISP_DIR	= $(FINAL_INSTALL_LIBDIR)/elisp
+FINAL_INSTALL_INFO_DIR	= $(infodir)
+FINAL_INSTALL_MAN_DIR	= $(mandir)
+FINAL_INSTALL_ELISP_DIR	= $(lispdir)
 FINAL_INSTALL_CGI_DIR	= @CGIDIR@
 
 # You should not need to override anything below here
-- 
2.1.2

